### PR TITLE
Helper print funcions

### PR DESCRIPTION
**Pull request by ghost**
```
Lines: 22
Columns: 78
-------
Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor
incididunt ut labore etdolore magna aliqua.

Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor
                            incididunt ut labore etdolore magna aliqua.
-------
Options:
      --config string      Location of config files (default
                           "/home/claudio/.john")blablabla blabla blabla
                           blabla bla bla blabla blabla bla bla blabla
                           balabalbaaa.
  -v, --version            Print version information and quit.
```

```
Lines: 22
Columns: 85
-------
Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor
incididunt ut labore etdolore magna aliqua.
Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor
                            incididunt ut labore etdolore magna aliqua.
Options:
      --config string      Location of config files (default
                           "/home/claudio/.john")blablabla blabla blabla blabla bla
                           bla blabla blabla bla bla blabla balabalbaaa.
  -v, --version            Print version information and quit.

```

```C
log_print(OUT_CONSOLE, VERB_DEFAULT, FALSE, strlen("Lorem ipsum dolor sit amet, "),
       "Lorem ipsum dolor sit amet, consectetur adipisicing"
       "elit, sed do eiusmod tempor incididunt ut labore et"
       "dolore magna aliqua. Value is %d.\n", 10);

```
```
Lines: 22
Columns: 90
-------
Options:
Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor incididunt
ut labore etdolore magna aliqua.
Lorem ipsum dolor sit amet, consectetur adipisicingelit, sed do eiusmod tempor incididunt
                            ut labore etdolore magna aliqua. Value is 10.
Options:
      --config string      Location of config files (default
                           "/home/claudio/.john")blablabla blabla blabla blabla bla bla
                           blabla blabla bla bla blabla balabalbaaa.
  -v, --version            Print version information and quit.
```

Usage (better to put heading as a parameter? or as `strlen("Lorem ipsum dolor sit amet, "),`):
```
log_print(OUT_CONSOLE, VERB_DEFAULT, FALSE, 
       "Lorem ipsum dolor sit amet, " // note the space
       "consectetur adipisicing elit, sed do "
       "eiusmod tempor incididunt ut labore et "
       "dolore magna aliqua. Value is %d.\n", 10);
```

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+#include "config.h"
+#include "john.h"
+#include "logger.h"
+#include "options.h"
+#include "util.h"
+
+static int is_separator(char symbol)
+{
+
+	if (symbol == ' ' || symbol == '\t' || symbol == '\n' || symbol == '\0')
```
Comment (ghost) [inline src/util.c]: More separators: `)`,`]`,`}`,`,` , `.`.

File: src/util.c [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	return len;
+}
+
+static int wrap(const char * text, char * line, const int width, int first_line,
+    int identation)
+{
+	int line_len = 0, start = 0, limit;
+
+	/* ltrim text. */
+	while (!first_line && text[start] != '\0' && is_separator(text[start]))
```
Comment (ghost) [inline src/util.c]: The correct here is `is_blank(char);`

Comment (magnumripper) [reply src/util.c]: Yes, or `is_whitespace()`

**Comment by magnumripper** [comment]
Ref #3511 

Cool! Maybe this new `util.c` is where this function should be, but in that case it shouldn't be called `log_print()` because the `log_` prefix is for `logger.c`.

I pictured a `log_print()` function in `logger.c` since it can optionally log and/or print the one same message (as-is to log and line wrapped to screen). I'm still thinking perhaps the lot should be in `logger.c` (@solardiz what's your opinion?).

File: src/util.c [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+		if (limit && *(cols) > limit)
+			*(cols) = limit;
+		if (*(cols) > 140)
+			*(cols) = 140;
+		if (*(cols) < 60)
+			*(cols) = 60;
+		return 0;
+	}
+#endif
+	*(lines) = 20;
```
Comment (magnumripper) [inline src/util.c:119]: I think default height would traditionally be 24.

**Comment by solardiz** [comment]
> I'm still thinking perhaps the lot should be in `logger.c` (@solardiz what's your opinion?).

I agree that at least a function that both logs to a file and prints to the terminal should be in `logger.c`. Possibly other related functions too. Also, I am unhappy with any of this being GPL-only. I am barely OK with Claudio's formats being GPL-only because they're easy to exclude, but this is shared code that wouldn't be so easy to exclude if need be for licensing reasons.

File: src/util.c (hunk 3 above)
Comment (ghost) [reply src/util.c:119]: I already changed this (to 24x80). The default in my (GNOME) terminal.

**Comment by ghost** [comment]
(After renaming it to `gpl_util.?`) I'm planning to use this for OpenCL messaging. I'll add a helper that prints the correct `device_id` (1-based) and controls `NODE` printing.

I will (in shared code):
```
#ifdef GPL
    fancy_print();
#else
    printf(...); //Regular
#endif
```
Colors:
image

**Pull request closed without merging**